src/snowflake/snowpark/_internal/xml_reader.py

```diff
@@ -399,8 +411,14 @@ def process_xml_range(
             record_str = re.sub(r"&(\w+);", replace_entity, record_str)
 
             try:
-                element = ET.fromstring(record_str)
-                yield element_to_dict(strip_namespaces(element))
+                if lxml_installed:
+                    parser = ET.XMLParser(recover=True, ns_clean=True)
+                    element = ET.fromstring(record_str, parser)
+                else:
+                    element = ET.fromstring(record_str)
+                if strip_namespaces:
+                    element = strip_xml_namespaces(element)
+                yield element_to_dict(element)
```

Comment:
I'm a little confused。
would there be any functional difference between when lxml is installed and lxml is not installed?

also is this part of code executed in UDTF? I saw you added 
```
packages=["snowflake-snowpark-python", "lxml"],
```

in that case, wouldn't lxml be guaranteed available?

Reply:
Yeah for the UDTF, lxml will always be there; but this is also a file in snowpark python, and lxml is only added as a dev dependency. Although it's very unlikely that users will import this module locally, if they do, it will fail due to import error. So, just added it for this reason.

Reply:
I see, I feel it's worth adding the reason as comments in the code
otherwise I would think the code would always work regardless of whether lxml_installed
```python
                if lxml_installed:
                    parser = ET.XMLParser(recover=True, ns_clean=True)
                    element = ET.fromstring(record_str, parser)
                else:
                    element = ET.fromstring(record_str)
```

Reply:
done, added comments